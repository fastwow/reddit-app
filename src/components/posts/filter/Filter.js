import {TextInput} from 'react-native';
import styles from './styles';
import PropTypes from 'prop-types';
import React from 'react';

const Filter = ({filter, onChangeText}) => {

  return (
    <TextInput
      selectionColor="#ffffff" style={styles.search}
      placeholder="Search"
      value={filter.searchTerm}
      onChangeText={onChangeText}/>
  );
};

Filter.propTypes = {
  filter: PropTypes.shape({searchTerm: PropTypes.string}).isRequired,
  onChangeText: PropTypes.func.isRequired,
};

export default Filter;
