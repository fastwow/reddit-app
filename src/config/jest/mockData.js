export const postsResponse = {
  children: [
    {
      data: {
        url: 'https://v.redd.it/1zhxx58rhp421',
        id: '98vj9e',
        thumbnail: 'https://b.thumbs.redditmedia.com/NFJ5zzCUnxTgAl71EkjSSCePc7wtcCLqTdbYgrwJkdU.jpg',
        title: 'Blinks of affection',
      },
    },
    {
      data: {
        url: 'https://i.imgur.com/GsoJi0d.gifv',
        id: 'a6q94b',
        thumbnail: 'https://b.thumbs.redditmedia.com/cjYmtjKiy9OT1y43RFMuTkcH_WmTM66RsSuN7c6xkhA.jpg',
        title: 'This church looks like a gingerbread house.',
      },
    },
  ],
};

export const posts = postsResponse.children.map(c => c.data);
