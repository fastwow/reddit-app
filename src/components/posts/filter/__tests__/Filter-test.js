import React from 'react';
import renderer from 'react-test-renderer';
import jest from 'jest-mock';
import Filter from '../Filter';

test('Filter renders correctly', () => {
  const tree = renderer.create(
    <Filter
      onChangeText={jest.fn()}
      filter={{searchTerm: 'cat'}}/>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
