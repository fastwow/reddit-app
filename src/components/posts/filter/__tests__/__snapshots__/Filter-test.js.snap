// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[MockFunction]}
  placeholder="Search..."
  style={
    Object {
      "borderColor": "#d6d7da",
      "borderRadius": 20,
      "borderWidth": 2,
      "color": "#000000",
      "fontSize": 20,
      "height": 40,
      "margin": 5,
      "textAlign": "center",
    }
  }
  underlineColorAndroid="transparent"
  value="cat"
/>
`;
